scripts/target.js

```diff
@@ -0,0 +1,281 @@
+      attributes: true,
+      attributeFilter: ['class'],
+    };
+    const observer = new MutationObserver(() => {
+      if (document.body.classList.contains('appear')) {
+        // eslint-disable-next-line no-console
+        console.debug('content has been decorated... resolving');
+        observer.disconnect();
+        resolve(document.body.querySelector('main'));
+      }
+    });
+    observer.observe(document.body, config);
+  });
+}
+
+/**
+ * Get all sections that are already loaded.
+ * @param main The main element.
+ */
+function getLoadedSections(main) {
+  const sections = main.querySelectorAll('.section');
+  return Array.from(sections)
+    .map((section) => new Promise((resolve) => {
+      if (section.getAttribute('data-section-status') === 'loaded') {
+        // eslint-disable-next-line no-console
+        console.debug('section is already loaded... resolving immediately', section);
+        resolve(section);
+      }
+      const config = {
+        attributes: true,
+        attributeFilter: ['data-section-status'],
+      };
+      const observer = new MutationObserver(() => {
+        if (section.getAttribute('data-section-status') === 'loaded') {
+          // eslint-disable-next-line no-console
+          console.debug('section has been loaded... resolving', section);
+          observer.disconnect();
+          resolve(section);
+        }
+      });
+      observer.observe(section, config);
+    }));
+}
+
+/**
+ * Render offers in a section.
+ * @param section The section.
+ * @param offers The offers.
+ */
+function displayOffers(section, offers) {
+  offers.forEach((offer) => {
+    const { type, cssSelector, content } = offer;
+    const targetElement = section.querySelector(escapeSelector(cssSelector));
+    if (targetElement) {
+      switch (type) {
+        case 'insertAfter':
+          console.debug('inserting content after', targetElement); // eslint-disable-line no-console
+          targetElement.insertAdjacentHTML('afterend', content);
+          break;
+        case 'insertBefore':
+          console.debug('inserting content before', targetElement); // eslint-disable-line no-console
+          targetElement.insertAdjacentHTML('beforebegin', content);
+          break;
+        case 'setHtml':
+          console.debug('setting content of', targetElement); // eslint-disable-line no-console
+          targetElement.innerHTML = content;
+          break;
+        default:
+          console.warn(`unsupported action type ${type}`); // eslint-disable-line no-console
+      }
+      console.debug('section has been rendered', section); // eslint-disable-line no-console
+      window?.measurePerformance(
+        `targeting:rendering-section:${Array.from(section.classList).join('_')}`,
+      );
+    }
+  });
+  if (section.style.visibility === 'hidden') {
+    // eslint-disable-next-line no-console
+    console.debug('revealing section', section);
+    section.style.visibility = 'visible';
```

Comment:
I think `visible` should be the default, so you could just set it to `null` or `''` and save a few bytes